### ISSUE TITLE
scale_fp8_row: unmasked column-scale load reads past the end of w_scale

### ISSUE THREAD

**Issue by truong-v** [pytorch/FBGEMM#6157]
## Summary

`_kernel_scale_fp8_row` masks its activation load and its store with `n_offset < N`, but loads the column scales with a bare `tl.load(w_scale + n_offset)`. `w_scale` has exactly `N` elements and the autotuner's smallest `BLOCK_SIZE` is 512, so every call with `N < 512` reads `BLOCK_SIZE - N` elements past the end of the tensor; for larger `N` the final loop iteration overruns whenever `N` is not a multiple of `BLOCK_SIZE`.

The store is masked, so results are unaffected and nothing crashes under the caching allocator — which is why this is easy to miss. It is still a real out-of-bounds read, and it is on by default in the test suite: `test_scale_fp8_row` calls the kernel at `M=2, N=3`, a 512-wide load over a 12-byte tensor.

https://github.com/pytorch/FBGEMM/blob/bef38467f88437baddad319c1c4ecba820c97e13/fbgemm_gpu/experimental/gemm/triton_gemm/fp8_gemm.py#L3071-L3082

## Reproduction

```python
import torch
from fbgemm_gpu.experimental.gemm.triton_gemm.fp8_gemm import scale_fp8_row

M, N = 2, 3  # the shape fp8_gemm_test.py:435 already uses
a = torch.randn(M, N, device="cuda", dtype=torch.bfloat16)
x_scale = torch.randn(M, device="cuda").abs()
w_scale = torch.randn(N, device="cuda").abs()  # 12 bytes, allocated last
scale_fp8_row(a, x_scale, w_scale)
torch.cuda.synchronize()
```

```
$ PYTORCH_NO_CUDA_MEMORY_CACHING=1 compute-sanitizer --tool memcheck python repro.py
========= Invalid __global__ read of size 16 bytes
=========     at _kernel_scale_fp8_row+0x6b0 in fp8_gemm.py:3076
=========     by thread (96,0,0) in block (0,0,0)
=========     Access to 0x7fee7e400c00 is out of bounds
=========     and is 1525 bytes after the nearest allocation at 0x7fee7e400600 of size 12 bytes
...
========= ERROR SUMMARY: 194 errors
```

The 12-byte allocation is `w_scale`; every reported address lies past its end, and with the caching allocator disabled the overrun also takes the context down with `unspecified launch failure`.

## Environment

- pytorch/FBGEMM at `bef38467f88437baddad319c1c4ecba820c97e13` (main)
- NVIDIA B200 (sm_100), driver 595.71.05
- torch 2.13.0+cu130, triton 3.7.1
- compute-sanitizer 2025.4.1 (CUDA 13.1)
